Alignment/OfflineValidation/plugins/PixelBaryCentreAnalyzer.cc

```diff
@@ -0,0 +1,603 @@
+/** \class PixelBaryCentreAnalyzer
+ *  The analyer works as the following :
+ *  - Read global tracker position from global tag
+ *  - Read tracker alignment constants from different ESsource with different labels
+ *  - Calculate barycentres for different pixel substructures using global tracker position and alignment constants and store them in trees, one for each ESsource label.
+ *
+ *  Python script plotBaryCentre_VS_BeamSpot.py under script dir is used to plot barycentres from alignment constants used in Prompt-Reco, End-of-Year Rereco and so-called Run-2 (Ultra)Legacy Rereco. Options of the plotting script can be found from the helper in the script.
+ *
+ *  $Date: 2021/01/05 $
+ *  $Revision: 1.0 $
+ *  \author Tongguang Cheng - Beihang University <tongguang.cheng@cern.ch>
+ *
+*/
+
+// Framework
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESWatcher.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+
+// Phase-1 Pixel
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "DataFormats/TrackerCommon/interface/PixelBarrelName.h"
+#include "DataFormats/TrackerCommon/interface/PixelEndcapName.h"
+
+// pixel quality
+#include "CondFormats/SiPixelObjects/interface/SiPixelQuality.h"
+#include "CondFormats/DataRecord/interface/SiPixelQualityFromDbRcd.h"
+// global postion
+#include "CondFormats/Alignment/interface/DetectorGlobalPosition.h"
+#include "CondFormats/AlignmentRecord/interface/GlobalPositionRcd.h"
+// tracker alignment
+#include "CondFormats/AlignmentRecord/interface/TrackerAlignmentRcd.h"
+#include "CondFormats/Alignment/interface/Alignments.h"
+// beamspot
+#include "CondFormats/DataRecord/interface/BeamSpotObjectsRcd.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+
+// TFileService
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+// ROOT
+#include <TTree.h>
+#include <TString.h>
+#include <TVector3.h>
```

Comment:
Based on the last available round of static analyzer checks this inclusion in problematic, see [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a64171/12069/llvm-analysis/report-276203.html#EndPath). 
I am wondering if this dependency on ROOT types could be avoid using either a C++ native type or a `cmssw` built-in type (such as `GlobalPoint`).

Reply:
What about triple the branches by storing explicit x, y, z coordinates?

Reply:
I don't think that `GlobalPoint` would work here because the vectors are added and multiplied but `GlobalVector` should be suitable.

Reply:
I was referring to the type stored in the tree. The vector algebra can be carried out by whatever means.

Reply:
Done